### DISCUSSION TITLE
Add options page to existing Treeview node in Tools/Options

### DISCUSSION THREAD

**Discussion by RobertvanderHulst** [VsixCommunity/Community.VisualStudio.Toolkit#416] [Q&A] [answered]
I would like to add an options page to an existing Treeview node in Tools/Options.
In our case, I would like to add Debugging/X# Debugger 
I think have specified the right names, but our entry appears as a separate entry:
![image](https://user-images.githubusercontent.com/14240939/218711579-f1b2bdea-37ea-40d0-bb26-4c6cdb9636a0.png)
Does anybody know how we can have our page appear under the main Debugging tree?

**Comment by reduckted** [comment, marked as answer]
I think the reason it's not appearing under the "Debugging" node is because the node's identity and the actual text displayed is different. The node's identity is `"Debugger"`, whereas the displayed text is coming from a resource file with the value "Debugging".

Here's how I discovered this:
1. Get the [Settings Store Explorer](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.SettingsStoreExplorer) extension if you don't already have it.
2. Open its tool window (_View -> Other Windows -> Settings Store Explorer_)
3. Expand `Config/ToolsOptionsPages`

You'll see that there's a `Debugger` node with `General`, `HotReload`, `JIT`, etc as children.

So if you define your options page with a `categoryName` of "Debugger", then it should appear in the correct place.

```cs
[ProvideOptionPage(typeof(MyOptionsPage), "Debugger", "X# Debugger", 0, 0, true)]
```

**Comment by RobertvanderHulst** [comment]
Dave, that is brilliant. Thanks for the tip.
Next time I will look there first!
